### PR TITLE
Added support for Modtronix NZ32-SC151 board, based on STM32L151RC

### PR DESCRIPTION
**Pull request by modtronix-com**
Tested MSC, CDC and HID interfaces, and all work.

### CODE REVIEW

**Comment by c1728p9** [comment]
Please accept the contributors agreement:
https://developer.mbed.org/contributor_agreement/

**Comment by c1728p9** [comment]
Please add your configuration to [SUPPORTED_CONFIGURATIONS](https://github.com/mbedmicro/DAPLink/blob/master/test/info.py#L115). Also, if you would like your project included in the next release please add it to PROJECT_RELEASE_INFO  as well.

**Comment by modtronix-com** [comment]
Thanks for quick reply! I have already accepted the contributor agreement in 2014. If I click on link, I get message:
"You accepted the Contributor Agreement on Wed 03 Dec 2014."

I added configuration to SUPPORTED_CONFIGURATIONS and PROJECT_RELEASE_INFO in test.py as requested.

**Comment by c1728p9** [comment]
Hi @modtronix-com, I looked for modtronix-com on mbed but couldn't find it. What account are you using?

File: test/info.py [hunk 1]
```diff
@@ -125,6 +126,7 @@
     (   0x7404,     'lpc11u35_6lowpan_borderrouterethernet_if', None,               'mbed-6LoWPAN-Border-Router-ETHERNET'   ),
     (   0xC000,     'lpc11u35_cocorico_if',                     None,               None                                    ), # TODO - set target to 'CoCo-ri-Co' when mbed-os supports this
     (   0x5500,     'lpc11u35_gr_peach_if',                     None,               'Renesas-GR-PEACH'                      ),
+    (   0x6660,     'lpc11u35_nz32_sc151_if',                   None,               'NZ32-SC151'                            ),
```
Comment (c1728p9) [inline test/info.py]: I don't see ```NZ32-SC151``` supported in mbed-os. To run testing on the target mbed-os support is needed. To allow testing now just set target to ```None``` and leave a comment to the right indicating what the name would be.

Comment (modtronix-com) [inline test/info.py]: The target name in mbed-os is "NZ32_SC151". It is listed in "targets.json" file here:
https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1337
The platform page has also already been added to mbed, but not public yet:
https://developer.mbed.org/platforms/nz32-sc151/
For example, to compile project via mbed-cli using mbed-os, the following works:
mbed compile -t GCC_ARM -m NZ32_SC151

Comment (modtronix-com) [inline test/info.py]: Our account on mbed is "modtronix":
https://developer.mbed.org/users/modtronix/

Comment (c1728p9) [inline test/info.py]: Ok cool, thanks for the clarification. Since that target exists in mbed-os you can leave it here.

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
Hi @modtronix-com, please rebase this PR to resolve conflicts

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by modtronix-com** [comment]
Hello, sorry for delay. I have been spending lots of time trying to improve my git skills, and learn how to implement your request:
"... please rebase this PR to resolve conflicts".
I was just about to finish it, but seems like "c1728p9" has just done a new commit for this branch:
"Merge branch 'master' into nz32-sc151".
Please let me know what I am supposed to do to get this pull request accepted? Do I still have to perform the requested rebase?

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
Hi @modtronix-com, I solved the merge conflicts, so no need to update this PR.

**Comment by modtronix-com** [comment]
Hi, is there anything else I have to do? Or, will this PR soon be accepted?

**Comment by c1728p9** [comment]
Hi @modtronix-com, the PR has been merged.

**Pull request merged**